File: semantickernel-experimental/src/main/java/com/microsoft/semantickernel/data/vectorsearch/queries/VectorSearchQuery.java
// Copyright (c) Microsoft. All rights reserved.
package com.microsoft.semantickernel.data.vectorsearch.queries;

import com.microsoft.semantickernel.data.vectorstorage.options.VectorSearchOptions;

import java.util.List;

public class VectorSearchQuery {
    private final VectorSearchQueryType queryType;
    private final VectorSearchOptions searchOptions;

    public VectorSearchQuery(VectorSearchQueryType queryType, VectorSearchOptions searchOptions) {
        this.queryType = queryType;
        this.searchOptions = searchOptions;
    }

    public VectorSearchQueryType getQueryType() {
        return queryType;
    }

    public VectorSearchOptions getSearchOptions() {
        return searchOptions;
    }

    public static VectorizedSearchQuery createQuery(List<Float> vector,
        VectorSearchOptions options) {
        return new VectorizedSearchQuery(vector, options);
    }

    public static VectorizableTextSearchQuery createQuery(String searchText,
        VectorSearchOptions options) {
        return new VectorizableTextSearchQuery(searchText, options);
    }
}

File: 01-core-implementations/python/semantic_kernel/data/vector_search_query.py
#!/usr/bin/env python3
"""
Vector Search Query module

Copyright (c) 2025 Bryan Roe
Licensed under the MIT License

This file is part of the Semantic Kernel - Advanced AI Development Framework.
Original work by Bryan Roe.

Author: Bryan Roe
Created: 2025
License: MIT
"""

# Copyright (c) Microsoft. All rights reserved.


from typing import Literal

from pydantic import Field

from semantic_kernel.data.const import VectorSearchQueryTypes
from semantic_kernel.data.vector_search_options import VectorSearchOptions
from semantic_kernel.kernel_pydantic import KernelBaseModel


class VectorSearchQuery(KernelBaseModel):
    """A query for vector search."""

    query_type: Literal[
        VectorSearchQueryTypes.VECTORIZED_SEARCH_QUERY,
        VectorSearchQueryTypes.VECTORIZABLE_TEXT_SEARCH_QUERY,
        VectorSearchQueryTypes.HYBRID_TEXT_VECTORIZED_SEARCH_QUERY,
        VectorSearchQueryTypes.HYBRID_VECTORIZABLE_TEXT_SEARCH_QUERY,
    ]
    search_options: VectorSearchOptions | None = Field(default_factory=VectorSearchOptions)
    query_text: str | None = None
    vector: list[float | int] | None = None
